source/solvers/navier_stokes_assemblers.cc

```diff
@@ -1712,16 +1712,16 @@ BuoyancyAssembly<dim>::assemble_rhs(
       // Current temperature values
       double current_temperature = scratch_data.temperature_values[q];
 
-      strong_residual[q] += -force * thermal_expansion * current_temperature;
+      strong_residual[q] += force * thermal_expansion * current_temperature;
```

Comment:
I don't think this is correct. What are the equations you are solving?
u . grad(u) = -grad(p) + mu lap(u) + beta * "force" * (T-T0) no?
In this case, force should have a minus sign in front of it. 
I suggest you take the opportunity to update the documentation to make that clear.

Reply:
The equation is: u . grad(u) = -grad(p) + mu lap(u) - beta * "force" * (T-T0)